main/src/vocab/nllb_vocab.rs

```diff
@@ -0,0 +1,245 @@
+    "hat_Latn", "hau_Latn", "heb_Hebr", "hin_Deva", "hne_Deva", "hrv_Latn", "hun_Latn", "hye_Armn",
+    "ibo_Latn", "ilo_Latn", "ind_Latn", "isl_Latn", "ita_Latn", "jav_Latn", "jpn_Jpan", "kab_Latn",
+    "kac_Latn", "kam_Latn", "kan_Knda", "kas_Arab", "kas_Deva", "kat_Geor", "knc_Arab", "knc_Latn",
+    "kaz_Cyrl", "kbp_Latn", "kea_Latn", "khm_Khmr", "kik_Latn", "kin_Latn", "kir_Cyrl", "kmb_Latn",
+    "kon_Latn", "kor_Hang", "kmr_Latn", "lao_Laoo", "lvs_Latn", "lij_Latn", "lim_Latn", "lin_Latn",
+    "lit_Latn", "lmo_Latn", "ltg_Latn", "ltz_Latn", "lua_Latn", "lug_Latn", "luo_Latn", "lus_Latn",
+    "mag_Deva", "mai_Deva", "mal_Mlym", "mar_Deva", "min_Latn", "mkd_Cyrl", "plt_Latn", "mlt_Latn",
+    "mni_Beng", "khk_Cyrl", "mos_Latn", "mri_Latn", "zsm_Latn", "mya_Mymr", "nld_Latn", "nno_Latn",
+    "nob_Latn", "npi_Deva", "nso_Latn", "nus_Latn", "nya_Latn", "oci_Latn", "gaz_Latn", "ory_Orya",
+    "pag_Latn", "pan_Guru", "pap_Latn", "pol_Latn", "por_Latn", "prs_Arab", "pbt_Arab", "quy_Latn",
+    "ron_Latn", "run_Latn", "rus_Cyrl", "sag_Latn", "san_Deva", "sat_Beng", "scn_Latn", "shn_Mymr",
+    "sin_Sinh", "slk_Latn", "slv_Latn", "smo_Latn", "sna_Latn", "snd_Arab", "som_Latn", "sot_Latn",
+    "spa_Latn", "als_Latn", "srd_Latn", "srp_Cyrl", "ssw_Latn", "sun_Latn", "swe_Latn", "swh_Latn",
+    "szl_Latn", "tam_Taml", "tat_Cyrl", "tel_Telu", "tgk_Cyrl", "tgl_Latn", "tha_Thai", "tir_Ethi",
+    "taq_Latn", "taq_Tfng", "tpi_Latn", "tsn_Latn", "tso_Latn", "tuk_Latn", "tum_Latn", "tur_Latn",
+    "twi_Latn", "tzm_Tfng", "uig_Arab", "ukr_Cyrl", "umb_Latn", "urd_Arab", "uzn_Latn", "vec_Latn",
+    "vie_Latn", "war_Latn", "wol_Latn", "xho_Latn", "ydd_Hebr", "yor_Latn", "yue_Hant", "zho_Hans",
+    "zho_Hant", "zul_Latn",
+];
+
+pub struct NLLBVocab {
+    /// A mapping of tokens as string to indices (i.e. the encoder base)
+    pub values: HashMap<String, i64>,
+
+    /// A mapping of token IDs to strings (i.e. the decoder base)
+    pub indices: HashMap<i64, String>,
+
+    /// A mapping of special value tokens as strings to IDs (i.e. the encoder base for special
+    /// values), special values typically include things like BOS/EOS markers, class markers, mask
+    /// markers and padding markers
+    pub special_values: HashMap<String, i64>,
+
+    /// A mapping of special value tokens as IDs to strings (i.e. the decoder base for special values)
+    pub special_indices: HashMap<i64, String>,
+
+    /// Language code stored as bytes for extraction of the prefix in input sequences
+    pub language_codes_bytes: HashSet<Vec<u8>>,
+
+    pub special_token_storage: SpecialTokenMap,
+}
+
+impl NLLBVocab {
+    /// The beginning of sequence token that was used during pretraining.
+    /// Can be used a sequence classifier token.
+    pub fn get_bos_value(&self) -> &str {
+        self.special_token_storage
+            .bos_token
+            .as_deref()
+            .unwrap_or("<s>")
```

Comment:
can you please align with https://github.com/guillaume-be/rust-tokenizers/blob/4d8aa3eb964fd33cb0dbc6bfd86b9edb5d264faf/main/src/vocab/m2m100_vocab.rs#L75 and declare the default constant separately - this would allow a clearer overview. Please also note it seems you are missing defaults for `cls`, `unk` and `mask`

Reply:
Fixed in https://github.com/npatsakula/rust-tokenizers/pull/5.